auth/src/main/java/com/vimeo/networking2/requests/clientcredentials/ClientCredentialsInteractor.kt

```diff
@@ -0,0 +1,90 @@
+package com.vimeo.networking2.requests.clientcredentials
+
+import com.vimeo.networking2.ApiError
+import com.vimeo.networking2.VimeoAccount
+import com.vimeo.networking2.enums.GrantType
+import com.vimeo.networking2.enums.ScopeType
+import com.vimeo.networking2.requests.ApiResponse
+import com.vimeo.networking2.requests.AuthCallback
+import com.vimeo.networking2.requests.AuthService
+import com.vimeo.networking2.utils.ErrorResponseConverter
+import com.vimeo.networking2.utils.awaitResponse
+import com.vimeo.networking2.utils.createApiErrorForCustomMessage
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import retrofit2.Response
+
+/**
+ * Interactor which performs auth request with client credentials.
+ *
+ * @param errorResponseConverter    Used to convert error response to [ApiError].
+ * @param authService               Retrofit service for all auth requests.
+ * @param authHeaders               Auth Header generated from the client id and client secret.
+ * @param apiScopes                 List of [ScopeType]s.
+ * @param authTokenCallback         Callback to inform you that the authentication was successful
+ *                                  and the access token can be set for every request.
+ */
+class ClientCredentialsInteractor(
+    private val errorResponseConverter: ErrorResponseConverter,
+    private val authService: AuthService,
+    private val authHeaders: String,
+    private val apiScopes: List<ScopeType>,
+    private val coroutineScope: CoroutineScope,
```

Comment:
Will java consumers be able to use this interactor? Perhaps we should have a separate `auth-coroutines` module